tools/src/main/scala/scala/scalanative/interflow/Interflow.scala

```diff
@@ -20,44 +20,75 @@ class Interflow(val config: build.Config)(implicit
     with PolyInline
     with Intrinsics
     with Log {
-  implicit val platform: PlatformInfo = PlatformInfo(config)
 
+  /** The target machine for which code is being compiled. */
+  implicit val target: TargetInfo = TargetInfo(config)
+
+  /** A map from symbol to its original definition. */
   private val originals = {
     val out = mutable.Map.empty[nir.Global, nir.Defn]
     analysis.defns.foreach { defn => out(defn.name) = defn }
     out
   }
 
+  /** The set of definitions to process. */
   private val todo = mutable.Queue.empty[nir.Global.Member]
+
+  /** A map from symbol to its optimized definition. */
   private val done = mutable.Map.empty[nir.Global.Member, nir.Defn.Define]
+
+  /** A set with the symbols for which Interflow has started. */
   private val started = mutable.Set.empty[nir.Global.Member]
+
+  /** A set with the symbols that shouldn't be processed. */
   private val denylist = mutable.Set.empty[nir.Global.Member]
+
+  /** A set with the symbols that are accessible from code roots. */
   private val reached = mutable.HashSet.empty[nir.Global.Member]
+
+  /** A map indicatting whether a particular symbol is pure. */
   private val modulePurity = mutable.Map.empty[nir.Global.Top, Boolean]
 
+  /** Returns the current scope. */
```

Comment:
This is kind of a performative comment. It would be nice if we could describe what a scope is more precisely.

Reply:
`ScopedVar` is a list a stack-like variable which can be appended when entering some scope and removed from the stack when existing the scope. Eg. 
```scala 
scoped(currentDefn := Foo):
  currentDefn.get // yields Foo
  scoped(currentDefn := Bar): 
    currentDefn.get // yields Bar
  currentDefn.get // yield.Foo  
```
These are uses to preserve current state when entering some nested area of code without usage of global variables and manually reasignnning them with old value. 

In case of currentFreshScope thats a `ThreadLocal[ScopedVar[nir.Fresh]]` so each thread has it's own stack of `nir.Fresh` generators. When it starts to inline a method it would create a new nir.Fresh on top of the stack and would use it in the MergeProcessor. When it's done it would pop this value.